pkg/querier/querier.go

```diff
@@ -0,0 +1,136 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package querier provides tools to interact with the Minder database
+package querier
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/mindersec/minder/internal/db"
+	"github.com/mindersec/minder/pkg/config/server"
+	"github.com/mindersec/minder/pkg/engine/selectors"
+	"github.com/mindersec/minder/pkg/eventer"
+	"github.com/mindersec/minder/pkg/profiles"
+	"github.com/mindersec/minder/pkg/ruletypes"
+)
+
+// Store interface provides functions to execute db queries and transactions
+type Store interface {
+	ProjectHandlers
+	RuleTypeHandlers
+	ProfileHandlers
+	BundleHandlers
+	BeginTx() (Querier, error)
+}
+
+// Querier interface provides functions to interact with the Minder database using transactions
+type Querier interface {
+	ProjectHandlers
+	RuleTypeHandlers
+	ProfileHandlers
+	BundleHandlers
+	CommitTx() error
+	CancelTx() error
+}
+
+// Type represents the querier type
+type Type struct {
+	store      db.Store
+	tx         *sql.Tx
+	querier    db.ExtendQuerier
+	ruleSvc    ruletypes.RuleTypeService
+	profileSvc profiles.ProfileService
+}
+
+// Ensure Type implements the Querier interface
+var _ Querier = (*Type)(nil)
+
+// New creates a new instance of the querier
+func New(ctx context.Context, config *server.Config) (Store, func(), error) {
```

Comment:
You may want to define a type for the second return argument (it's not clear what you do with a `func()` on its own).